yt/fields/magnetic_field.py

```diff
@@ -170,35 +170,56 @@ def _mach_alfven(field,data):
 
 def setup_magnetic_field_aliases(registry, ds_ftype, ds_fields, ftype="gas"):
     r"""
-    This routine sets up special aliases between dataset-specific magnetic fields
-    and the default magnetic fields in yt so that unit conversions between different
-    unit systems can be handled properly. This is only called from the `setup_fluid_fields`
-    method of a frontend's :class:`FieldInfoContainer` instance.
-
+    This routine sets up special aliases between dataset-specific magnetic
+    fields and the default magnetic fields in yt so that unit conversions
+    between different unit systems can be handled properly. This is only called
+    from the `setup_fluid_fields` method (for grid dataset) or the
+    `setup_gas_particle_fields` method (for particle dataset) of a frontend's
+    :class:`FieldInfoContainer` instance.
     Parameters
     ----------
     registry : :class:`FieldInfoContainer`
         The field registry that these definitions will be installed into.
     ds_ftype : string
-        The field type for the fields we're going to alias, e.g. "flash", "enzo", "athena", etc.
-    ds_fields : list of strings
-        The fields that will be aliased.
+        The field type for the fields we're going to alias, e.g. "flash",
+        "enzo", "athena", "PartType0", etc.
+    ds_fields : list of strings or string
+        The fields that will be aliased. For grid dataset, this should be a
+        list of strings corresponding to the components of magnetic field. For
+        particle dataset, this should be a single string corresponding to the
+        vector magnetic field.
     ftype : string, optional
         The resulting field type of the fields. Default "gas".
-
     Examples
     --------
+    >>> from yt.fields.magnetic_field import setup_magnetic_field_aliases
     >>> class PlutoFieldInfo(ChomboFieldInfo):
     ...     def setup_fluid_fields(self):
-    ...         from yt.fields.magnetic_field import \
-    ...             setup_magnetic_field_aliases
-    ...         setup_magnetic_field_aliases(self, "chombo", ["bx%s" % ax for ax in [1,2,3]])
+    ...         setup_magnetic_field_aliases(
+    ...             self, "chombo", ["bx%s" % ax for ax in [1,2,3]]
+    ...         )
+    >>> class GizmoFieldInfo(GadgetFieldInfo):
+    ...     def setup_gas_particle_fields(self):
+    ...         setup_magnetic_field_aliases(
+    ...             self, "PartType0", "MagneticField", ftype="PartType0"
+    ...         )
     """
     unit_system = registry.ds.unit_system
-    ds_fields = [(ds_ftype, fd) for fd in ds_fields]
-    if ds_fields[0] not in registry:
+    if isinstance(ds_fields, list):
+        # If ds_fields is a list, we assume a grid dataset
+        sampling_type = "cell"
+        ds_fields = [(ds_ftype, fd) for fd in ds_fields]
+        ds_field = ds_fields[0]
+    else:
+        # Otherwise, we assume a particle dataset
+        sampling_type = "particle"
+        ds_field = ds_fields
```

Comment:
I'm not sure you need this line, as it's getting clobbered in the next line.

Reply:
I agree it would be cleaner that way.